### PR TITLE
Umar/add documentation

### PR DESCRIPTION
**Pull request by mumarkhan999**
This pr is related to [https://edlyio.atlassian.net/browse/EDS-202](https://edlyio.atlassian.net/browse/EDS-202) story.

It includes the required steps and configurations to setup codecast.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,106 @@
...
-If not using oauth2, remove keys "database", "session" and "auth" and
-fill in settings in "configs" and "tokens".
+2. Checkout to `edly-master` branch
 
-If using oauth2, fill in settings in "database", "session" and "auth",
-and remove keys "configs" and "tokens".  Use oauth2_schema.sql, and
-add rows in user_configs where value is a json object with keys
-"s3AccessKeyId", "s3SecretAccessKey", "s3Region", "s3Bucket", and
-"uploadPath".  The row with user_id 0 is used for guest settings.
+        git checkout edly-master
```
Comment (HamzaIbnFarooq) [inline README.md]: don't we have to change this?

File: README.md [hunk 2]
```diff
@@ -1,35 +1,106 @@
...
-    npm start
+        mysql -u <YOUR MYSQL USER> -p
+
+    1. Inside mysql shell run
+
+            CREATE DATABASE <YOUR DATABASE NAME>;
+            USE <YOUR DATABASE NAME>;
+            SOURCE user_management_schema.sql;
+
+    2. Add `s3` bucket credentials run
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
    2. To add `s3` bucket credentials, run:
```

File: README.md [hunk 3]
```diff
@@ -1,35 +1,106 @@
...
+
+            INSERT INTO buckets (bucket_id, value) values (0,'{ \"s3AccessKeyId\": \"<YOUR S3 ACCESS KEY ID>\", \"s3SecretAccessKey\": \"<YOUR S3 SECRET ACCESS KEY>\", \"s3Region\": \"<YOUR S3 REGION>\", \"s3Bucket\": \"<YOUR S3 BUCKET NAME>\", \"uploadPath\": \"uploads\" }');
+
+    3. Add `admin` user run
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
    3. To add `admin` user, run
```

File: README.md [hunk 4]
```diff
@@ -1,35 +1,106 @@
...
 
-Start with these commands:
+3. At root level of the project run the following commands to create a database
 
-    npm install
-    npm run build
-    npm start
+        mysql -u <YOUR MYSQL USER> -p
+
+    1. Inside mysql shell run
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
    1. To create database, run
```

**Review by HamzaIbnFarooq — CHANGES_REQUESTED** [review]
little changes need to be done and kindly don't forget to sqash

File: README.md (hunk 3 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md (hunk 2 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md (hunk 1 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md (hunk 4 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md [hunk 5]
```diff
@@ -1,35 +1,102 @@
...
+
+    You might get an error something like that
+
+        ../deps/mpg123/src/output/alsa.c:19:10: fatal error: alsa/asoundlib.h: No such file or directory #include <alsa/asoundlib.h>
+
+    You can get rid of the above mentioned error by installing the following package
+
+        sudo apt-get install libasound2-dev
+
+    After you get rid of this error run `npm install` command again and if you get any error something like that is given below, just ignore it and proceed to next step
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
    After that, run command `npm install`  again and if you get any error similar to the one given below, just ignore it and proceed to next step
```

File: README.md [hunk 6]
```diff
@@ -1,35 +1,102 @@
...
+                }
+            },
+            "enableOauth": false
+        }
+
+4. Run the following command in terminal:
+
+        npm install
+
+    You might get an error something like that
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
    You might see an error similar to this
```

File: README.md [hunk 7]
```diff
@@ -1,35 +1,102 @@
...
+        gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
+        gyp ERR! System Linux 4.15.0-20-generic
+        gyp ERR! command "/home/umar/.nvm/versions/node/v12.10.0/bin/node" "/home/umar/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
+        gyp ERR! cwd /home/umar/Documents/Work/codecast/node_modules/speaker
+        gyp ERR! node -v v12.10.0
+        gyp ERR! node-gyp -v v5.0.3
+        gyp ERR! not ok
+
+
+5. Run the following commands
```
Comment (HamzaIbnFarooq) [inline README.md]: there is no difference in point 4's and 5's heading, can we make it more precise?

File: user_management_schema.sql [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      `email_id` varchar(225) NOT NULL,
+      `password` varchar(33) NOT NULL,
+      `is_active` BOOLEAN DEFAULT 1,
+      `is_admin` BOOLEAN DEFAULT 0,
+      `bucket_id` int(11) DEFAULT 0 NOT NULL,
+      PRIMARY KEY (`id`)
+    ) ENGINE=InnoDB  DEFAULT CHARSET=latin1 AUTO_INCREMENT=4;
+  
+  CREATE TABLE `buckets` (
+      `bucket_id` int(11) NOT NULL,
```
Comment (HamzaIbnFarooq) [inline user_management_schema.sql:21]: Let's make it unique, what do you say?

File: user_management_schema.sql [hunk 9]
```diff
@@ -0,0 +1,36 @@
+CREATE TABLE `sessions` (
+    `session_id` varchar(128) NOT NULL,
+    `expires` int(11) unsigned NOT NULL,
+    `data` text,
+    PRIMARY KEY (`session_id`)
+  ) DEFAULT CHARSET=utf8;
+  
+  CREATE TABLE IF NOT EXISTS `users` (
+      `id` int(5) NOT NULL AUTO_INCREMENT,
+      `email_id` varchar(225) NOT NULL,
```
Comment (HamzaIbnFarooq) [inline user_management_schema.sql:10]: We had made emai_id unique, didn't we? I guess the schema is not updated, kindly check.

File: user_management_schema.sql [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      PRIMARY KEY (`id`)
+    ) ENGINE=InnoDB  DEFAULT CHARSET=latin1 AUTO_INCREMENT=4;
+  
+  CREATE TABLE `buckets` (
+      `bucket_id` int(11) NOT NULL,
+      `value` text NOT NULL,
+      PRIMARY KEY (`bucket_id`)
+    ) DEFAULT CHARSET=utf8;
+  
+  CREATE TABLE IF NOT EXISTS `records` (
```
Comment (HamzaIbnFarooq) [inline user_management_schema.sql:26]: kindly use `IF NOT EXISTS` with all of them, for consistency.

File: user_management_schema.sql (hunk 9 above)
Comment (HamzaIbnFarooq) [reply user_management_schema.sql:10]: user doesn't seem to have username here, need to fix this schema.

**Review by HamzaIbnFarooq — CHANGES_REQUESTED** [review]
second iteration

File: user_management_schema.sql (hunk 8 above)
Comment (mumarkhan999) [reply user_management_schema.sql:21]: This PR is only related to the readme. There are no validations regarded to this in our code base so I think for we should leave as it is.

File: user_management_schema.sql (hunk 10 above)
Comment (mumarkhan999) [reply user_management_schema.sql:26]: done.

File: user_management_schema.sql (hunk 9 above)
Comment (mumarkhan999) [reply user_management_schema.sql:10]: done.

File: README.md (hunk 7 above)
Comment (mumarkhan999) [reply README.md]: It looks fine to me. If you have suggestions related to this, share it.

File: user_management_schema.sql (hunk 8 above)
Comment (mumarkhan999) [reply user_management_schema.sql:21]: bucket id is primary key for that table so no need to set it unique.

File: README.md (hunk 6 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md (hunk 5 above)
Comment (mumarkhan999) [reply README.md]: done.

File: README.md [hunk 11]
```diff
@@ -1,35 +1,102 @@
...
+                "saveUninitialized": true,
+                "cookie": {
+                    "secure": false,
+                    "maxAge": 60480000
+                }
+            },
+            "enableOauth": false
+        }
+
+4. Run the following command in terminal
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
4. To install its dependencies, run
```

File: README.md [hunk 12]
```diff
@@ -1,35 +1,102 @@
...
+        gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
+        gyp ERR! System Linux 4.15.0-20-generic
+        gyp ERR! command "/home/umar/.nvm/versions/node/v12.10.0/bin/node" "/home/umar/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
+        gyp ERR! cwd /home/umar/Documents/Work/codecast/node_modules/speaker
+        gyp ERR! node -v v12.10.0
+        gyp ERR! node-gyp -v v5.0.3
+        gyp ERR! not ok
+
+
+5. Run the following commands
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
5. To compile assets, run
```

File: README.md [hunk 13]
```diff
@@ -1,35 +1,102 @@
...
 
-Build `c-to-json` and copy the executable at the root of this project.
+1. Clone this repo using following command
 
-Copy `config.json.template` to `config.json` and edit it.
+        git clone https://github.com/edly-io/codecast.git
 
-If not using oauth2, remove keys "database", "session" and "auth" and
-fill in settings in "configs" and "tokens".
+2. At root level of the project run the following commands to create a database
```
Comment (HamzaIbnFarooq) [inline README.md]:
```suggestion
2. At root directory of the project, run the following commands to create a database
```

**Review by HamzaIbnFarooq — APPROVED** [review]

**Pull request merged**